tensorflow/python/keras/layers/wrappers_test.py

```diff
@@ -472,6 +472,54 @@ def test_TimeDistributed_set_static_shape(self):
     # Make sure the batch dim is not lost after array_ops.reshape.
     self.assertListEqual(outputs.shape.as_list(), [1, None, 30, 30, 16])
 
+  @keras_parameterized.run_all_keras_modes
+  def test_TimeDistributed_with_mimo(self):
+    class TestLayer(keras.layers.Layer):
+
+      def __init__(self):
+        super(TestLayer, self).__init__()
+        self.dense_1 = keras.layers.Dense(8)
+        self.dense_2 = keras.layers.Dense(16)
+
+      def build(self, input_shape):
+        super(TestLayer, self).build(input_shape)
+
+      def call(self, inputs):
+        return self.dense_1(inputs[0]), self.dense_2(inputs[1])
+
+      def compute_output_shape(self, input_shape):
+        output_shape_1 = self.dense_1.compute_output_shape(input_shape[0])
+        output_shape_2 = self.dense_2.compute_output_shape(input_shape[1])
+        return output_shape_1, output_shape_2
+
+    np.random.seed(100)
+    layer = TestLayer()
+
+    data_1 = array_ops.constant(
+        [[[[1.0], [1.0]], [[2.0], [2.0]]],
+         [[[4.0], [4.0]], [[5.0], [5.0]]],
+         [[[7.0], [7.0]], [[8.0], [8.0]]]])
+
+    data_2 = array_ops.constant(
+        [[[[1.0], [1.0]], [[2.0], [2.0]]],
+         [[[4.0], [4.0]], [[5.0], [5.0]]],
+         [[[7.0], [7.0]], [[8.0], [8.0]]]])
+
+    x1 = keras.Input(shape=(None, 2, 1), dtype='float32')
+    x2 = keras.Input(shape=(None, 2, 1), dtype='float32')
+    y1, y2 = keras.layers.TimeDistributed(layer)([x1, x2])
+    model_1 = keras.models.Model([x1, x2], [y1, y2])
+    model_1._run_eagerly = testing_utils.should_run_eagerly()
```

Comment:
This is not needed and discouraged since it is trying to access a private field. What you could do is using model.compile(run_eagerly=testing_utils.should_run_eagerly())